addons/mail/static/src/core/common/message.js

```diff
@@ -391,6 +391,19 @@ export class Message extends Component {
                 this.props.thread
             );
         }
+        if (oeType === "notification" && oeModel && oeId) {
+            ev.preventDefault();
+            await this.env.services.action.doAction({
+                type: "ir.actions.act_window",
+                res_model: oeModel,
+                views: [[false, "form"]],
+                res_id: Number(oeId),
+            });
+            if (!this.env.isSmall) {
+                this.props.thread.open(true, { autofocus: false });
+            }
+            return;
```

Comment:
There's no purpose of this `thread.open()`, the doAction above already does opening the thread.

But if you make use of `mail/message/<id:integer>` route then that's all you need